lib/urbanopt/geojson.rb

```diff
@@ -50,14 +54,322 @@ def measures_dir
+      # - min_elevation: number indicating minimum elevation across all buildings
+      # - mix_elevation: number indicating maximum elevation across all buildings
+      # - model: instance of OpenStudio::Model::Model
+      # - origin_lat_lon: instance of OpenStudio::PointLatLon indicating origin lat & lon
+      # - zoning: zoning is true if you'd like to utilize aspects of function that are specific to zoning
+        geometry = building_json[:geometry]
+        properties = building_json[:properties]
+        if zoning
+          source_id = properties[:source_id]
+        else
+          name = properties[:name]
+        end
+        floor_prints = []
+        multi_polygons = get_multi_polygons(building_json)
+        multi_polygons.each do |multi_polygon|
+          if multi_polygon.size > 1
+            runner.registerWarning("Ignoring holes in polygon")
+          end
+          multi_polygon.each do |polygon|
+            floor_print = floor_print_from_polygon(polygon, min_elevation, origin_lat_lon, runner, zoning)
+            if floor_print
+              floor_prints << floor_print
+            else
+              runner.registerWarning("Cannot get floor print for building '#{name}'")
+            end
+            break
+          end
+        end
+        result = []
+        floor_prints.each do |floor_print|
+          space = OpenStudio::Model::Space.fromFloorPrint(floor_print, max_elevation-min_elevation, model)
+          if space.empty?
+            runner.registerWarning("Cannot create building space")
+            next
+          end
+          space = space.get
+          space.setName("Building #{name} Space")
+          thermal_zone = OpenStudio::Model::ThermalZone.new(model)
+          thermal_zone.setName("Building #{name} ThermalZone")
+          space.setThermalZone(thermal_zone)
+          result << space
+        end
+        return result
+      end
+
+
+      def create_space_type(bldg_use, space_use, model)
+        # HELPER FUNCTION (or module method)
+      ##
+      # Returns instance of OpenStudio::Model::SpaceType
+      # NOTE: update this return value once test is made more specific
+      #
+      # Params:
+      # - bldg_use: string indicating building use (UPDATE THIS)
+      # - space_use: string indicating space use (UPDATE THIS)
+      # - model: instance of OpenStudio::Model::Model
+        name = "#{bldg_use}:#{space_use}"
+        # check if we already have this space type
+        model.getSpaceTypes.each do |s|
+          if s.name.get == name
+            return s
+          end
+        end
+        space_type = OpenStudio::Model::SpaceType.new(model)
+        space_type.setName(name)
+        space_type.setStandardsBuildingType(bldg_use)
+        space_type.setStandardsSpaceType(space_use)
+        return space_type
+      end
+
+
+      def create_other_buildings(building_json, surrounding_buildings, model, origin_lat_lon, runner)
```

Comment:
Maybe the functions that deal with OpenStudio Model and ModelObjects (Spaces, etc) should go in one module?  Functions that dont need OpenStudio stuff could be in another?  Just an idea for dividing things up